src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsService.kt

```diff
@@ -0,0 +1,98 @@
+package uk.gov.justice.digital.hmpps.prisonertonomisupdate.nonassociations
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.microsoft.applicationinsights.TelemetryClient
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.config.trackEvent
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.nomissync.model.CreateNonAssociationRequest
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.nonassociations.model.LegacyNonAssociation
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.CreateMappingRetryMessage
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.CreateMappingRetryable
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.NomisApiService
+import uk.gov.justice.digital.hmpps.prisonertonomisupdate.services.synchronise
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@Service
+class NonAssociationsService(
+  private val nonAssociationsApiService: NonAssociationsApiService,
+  private val nomisApiService: NomisApiService,
+  private val mappingService: NonAssociationMappingService,
+  private val nonAssociationsUpdateQueueService: NonAssociationsUpdateQueueService,
+  private val telemetryClient: TelemetryClient,
+  private val objectMapper: ObjectMapper,
+) : CreateMappingRetryable {
+
+  suspend fun createNonAssociation(event: NonAssociationDomainEvent) {
+    synchronise {
+      name = "nonAssociation"
+      telemetryClient = this@NonAssociationsService.telemetryClient
+      retryQueueService = nonAssociationsUpdateQueueService
+      eventTelemetry = mapOf("nonAssociationId" to event.additionalInformation.id.toString())
+
```

Comment:
@steverendell Will need to check that the NA hasn't been created in NOMIS, given we have 2-way sync I guess we would have this scenario:

1. NA created in NOMIS
2. NA Synchronised to DPS
3. DPS raises a create event 
And in this situation we need to ignore the create (and BTW the mapping may not be created yet since he sync has only just called the DPS endpoint)

For Sentence Adjustments we asked them to add something to message to indicate it was created by NOMIS so we can ignore the domain event